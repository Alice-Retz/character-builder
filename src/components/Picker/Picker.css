.selector {
	display: flex;
	flex-direction: column;
	align-items: flex-start;
	justify-content: space-around;
	margin-right: 10px;
}
