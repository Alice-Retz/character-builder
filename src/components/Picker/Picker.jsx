import React from "react";

export default function Picker({
	head,
	onHeadChange,
	middle,
	onMiddleChange,
	newCatchphrase,
	setCatchphrase,
	handleClick,
}) {
	const headOptions = ["boar", "wolf", "deer", "elephant"];
	const middleOptions = ["blue", "green", "brown"];

	return (
		<div>
			<label>
				Head
				<select value={head} onChange={(e) => onHeadChange(e.target.value)}>
					{headOptions.map((pic) => (
						<option key={pic}>{pic}</option>
					))}
				</select>
			</label>
			<label>
				Middle
				<select value={middle} onChange={(e) => onMiddleChange(e.target.value)}>
					{middleOptions.map((pic) => (
						<option key={pic}>{pic}</option>
					))}
				</select>
			</label>

			<label>
				New Catchphrase:
				<input
					type='text'
					value={newCatchphrase}
					onChange={(e) => setCatchphrase(e.target.value)}
				/>
			</label>
			<button type='button' onClick={handleClick}>
				Add catchphrase!
			</button>
		</div>
	);
}
