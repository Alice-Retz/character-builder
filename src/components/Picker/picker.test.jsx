import { screen, render } from "@testing-library/react";
import Picker from "./Picker";

it("should render our picker component", () => {
	const { container } = render(<Picker />);
	const selectHead = screen.getByLabelText(/Head/);
	const selectMiddle = screen.getByLabelText(/Middle/);
	const input = screen.getAllByLabelText(/New Catchphrase:/);

	expect(selectHead).toBeInTheDocument();
	expect(selectMiddle).toBeInTheDocument();
	expect(input).toBeInTheDocument();
	expect(container).toMatchSnapshot();
});
