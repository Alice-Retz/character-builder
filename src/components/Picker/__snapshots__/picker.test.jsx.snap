// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render our picker component 1`] = `
<div>
  <div
    class="selector"
  >
    <label>
      Head
      <select>
        <option>
          boar
        </option>
        <option>
          wolf
        </option>
        <option>
          deer
        </option>
        <option>
          elephant
        </option>
      </select>
    </label>
    <label>
      Middle
      <select>
        <option>
          blue
        </option>
        <option>
          green
        </option>
        <option>
          brown
        </option>
      </select>
    </label>
    <label>
      Bottom
      <select>
        <option>
          blue
        </option>
        <option>
          green
        </option>
        <option>
          brown
        </option>
      </select>
    </label>
    <label>
      New Catchphrase:
      <input
        type="text"
        value=""
      />
      <p />
      <button
        type="button"
      >
        Add catchphrase!
      </button>
    </label>
  </div>
</div>
`;
